lib/ansible/modules/network/fortios/fortios_config.py

```
@@ -0,0 +1,229 @@
+    if not HAS_PYFG:
+        module.fail_json(msg='Could not import the python library pyFG required by this module')
+
+    #define device
+    f = FortiOS( module.params['host'],
+        username=module.params['username'],
+        password=module.params['password'],
+        timeout=module.params['username'],
+        vdom=module.params['vdom'])
+
+    #connect
+    try:
+        f.open()
+    except:
+        module.fail_json(msg='Error connecting device')
+
+    #get  config
+    try:
+        f.load_config(path=module.params['filter'])
+        result['running_config'] = f.running_config.to_text()
+
+    except:
+        module.fail_json(msg='Error reading running config')
+
+    if module.params['state'] == 'backuped':
+        #backup
+        try:
+            backup_file = open(module.params['dest'], "w")
+            backup_file.write(result['running_config'])
+            backup_file.close()
+        except:
+            module.fail_json(msg='Error writing running config to file {0}'.format(module.params['dest']))
+    else:
+        #update config
+        #store config in str
+        try:
+            conf_str = open(module.params['src'], 'r').read()
+            f.load_config(in_candidate=True, config_text=conf_str)
+        except:
+            module.fail_json(msg="Can't open configuration file, or configuration invalid")
+
+        #get updates lines
+        change_string = f.compare_config()
+
+        #remove not updatable parts
+        c = FortiConfig()
+        c.parse_config_output(change_string)
+
+        for o in NOT_UPDATABLE_CONFIG_OBJECTS:
+            c.del_block(o)
+
+        change_string = c.to_text()
+
+        if change_string != "":
+            result['change_string'] = change_string
+            result['changed'] = True
+
+        #Commit if not check mode
+        if module.check_mode is False and change_string != "":
+            try:
+                f.commit(change_string)
+            except Exception:
```

catch all exception should be avoided